ext/MetalExt.jl

```diff
@@ -0,0 +1,17 @@
+module MetalExt
+    using KernelAbstractions
+    using Metal, MetalKernels, Adapt
+    import DiffEqGPU
+
+    DiffEqGPU.maxthreads(::MetalDevice) = 256
+    DiffEqGPU.maybe_prefer_blocks(::MetalDevice) = MetalDevice()
+
+    # TODO move to KA
+    Adapt.adapt_storage(::CPU, a::MtlArray) = adapt(Array, a)
```

Comment:
I have already opened a PR for this for other backends: https://github.com/JuliaGPU/KernelAbstractions.jl/pull/353

Reply:
Yeah I will try and have that in the 0.9 release